ansible_collections/arista/avd/plugins/modules/eos_cli_config_gen.py

```diff
@@ -0,0 +1,92 @@
+    type: str
+    required: true
+  config_filename:
+    description: The path to save the generated config to.
+    type: str
+    required: true
+  documentation_filename:
+    description: The path to save the generated documentation to if generate_device_doc is true.
+    type: str
+    required: true
+  read_structured_config_from_file:
+    description: Flag to indicate if the structured config should be read from a file or not.
+    type: bool
+    default: true
+  generate_device_doc:
+    description: Flag to generate the device documentation.
+    type: bool
+    default: true
+  device_doc_toc:
+    description: Flag to generate the table of content for the device documentation.
+    type: bool
+    default: true
+  conversion_mode:
+    description:
+      - Run data conversion in either "error", "warning", "info", "debug", "quiet" or "disabled" mode.
+      - Conversion will perform type conversion of input variables as defined in the schema.
+      - Conversion is intended to help the user to identify minor issues with the input data, while still allowing the data to be validated.
+      - During conversion, messages will be generated with information about the host(s) and key(s) which required conversion.
+      - conversion_mode:disabled means that conversion will not run.
+      - conversion_mode:error will produce error messages and fail the task.
+      - conversion_mode:warning will produce warning messages.
+      - conversion_mode:info will produce regular log messages.
+      - conversion_mode:debug will produce hidden messages viewable with -v.
+      - conversion_mode:quiet will not produce any messages.
+    required: false
+    default: "debug"
+    type: str
+    choices: [ "error", "warning", "info", "debug", "quiet", "disabled" ]
+  validation_mode:
+    description:
+      - Run validation in either "error", "warning", "info", "debug" or "disabled" mode.
+      - Validation will validate the input variables according to the schema.
+      - During validation, messages will be generated with information about the host(s) and key(s) which failed validation.
+      - validation_mode:disabled means that validation will not run.
+      - validation_mode:error will produce error messages and fail the task.
+      - validation_mode:warning will produce warning messages.
+      - validation_mode:info will produce regular log messages.
+      - validation_mode:debug will produce hidden messages viewable with -v.
+    required: false
+    default: "warning"
+    type: str
+    choices: [ "error", "warning", "info", "debug", "disabled" ]
+  cprofile_file:
+    description:
+      - Filename for storing cprofile data used to debug performance issues.
+      - Running cprofile will slow down performance in it self, so only set this while troubleshooting.
+    required: false
+    type: str
+"""
+
+EXAMPLES = r"""
+---
+- name: TODO TODO TODO Generate device configuration in structured format
```

Comment:
We need to remember this.

Reply:
[3ca53c1](https://github.com/aristanetworks/avd/pull/4117/commits/3ca53c1d0c6733fae6b0f44719e5d9dbd299b0b6)